### PR TITLE
cmd/lncli: add --with flag to list channels

### PR DESCRIPTION
**Pull request by cfromknecht**
This commit adds a flag to `listchannels` that filters by remote pubkey.

### CODE REVIEW

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -1990,6 +1990,12 @@ var listChannelsCommand = cli.Command{
 			Name:  "private_only",
 			Usage: "only list channels which are currently private",
 		},
+		cli.StringFlag{
+			Name: "with",
```
Comment (bhandras) [inline cmd/lncli/commands.go]: `with_peer` may be a better name, wdyt?

Comment (cfromknecht) [reply cmd/lncli/commands.go]: yeah i considered a more descriptive name but then thought it sounded nice when read like a sentence. i don't have a strong opinion either way, i'll let others chime in with their preference and then go with herd 👍

Comment (bhandras) [reply cmd/lncli/commands.go]: sgtm

**Comment by cfromknecht** [comment]
we don't typically add tests for lncli since it's predominately a debugging tool. I have tested against my node though:
 - `list channels --with`

File: cmd/lncli/commands.go (hunk 1 above)
Comment (joostjager) [reply cmd/lncli/commands.go]: I would do `--peer`. The "with-" prefix isn't what we usually do.

File: cmd/lncli/commands.go [hunk 2]
```diff
@@ -1999,6 +2005,20 @@ func listChannels(ctx *cli.Context) error {
...
+	// Assert the with filter is empty or 66 bytes in length.
+	withPeer := ctx.String("with")
+	if len(withPeer) > 0 && len(withPeer) != 66 {
+		return fmt.Errorf("--with pubkey length of %d is invalid, "+
+			"must be 0 or 66 bytes", len(withPeer))
+	}
+
+	// Assert the with filter is hex-encoded, the empty string trivially
+	// passes.
+	_, err := hex.DecodeString(withPeer)
```
Comment (joostjager) [inline cmd/lncli/commands.go]: `route.NewVertexFromStr` ?

**Comment by cfromknecht** [comment]
> Why the decision to do this client side, while other filtering is taking place server side?

@joostjager didn't seem performance critical, plus rpc users can do this fairly easily. regardless, changed it to filter internally, ptal.

File: cmd/lncli/commands.go (hunk 1 above)
Comment (cfromknecht) [reply cmd/lncli/commands.go]: done

File: lnrpc/rpc.proto [hunk 3]
```diff
@@ -1385,6 +1385,12 @@ message ListChannelsRequest {
     bool inactive_only = 2;
     bool public_only = 3;
     bool private_only = 4;
+
+    /**
+    Filters the response for channels with a target peer's pubkey. If with_peer
```
Comment (joostjager) [inline lnrpc/rpc.proto]: s/with_peer/peer

File: rpcserver.go [hunk 4]
```diff
@@ -2948,6 +2948,11 @@ func (r *rpcServer) ListChannels(ctx context.Context,
 			"`private_only` can be set, but not both")
 	}
 
+	if len(in.Peer) > 0 && len(in.Peer) != 33 {
```
Comment (joostjager) [inline rpcserver.go]: Could use `route.NewVertexFromBytes`

**Pull request merged**